public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -1025,12 +1025,26 @@ export function getFieldConfig(
 }
 
 export function makeTempoLink(title: string, serviceName: string, spanName: string, datasourceUid: string) {
-  let query = { queryType: 'nativeSearch' } as TempoQuery;
+  let query: TempoQuery = { refId: 'A', queryType: 'traceqlSearch', filters: [] };
   if (serviceName !== '') {
-    query.serviceName = serviceName;
+    query.filters.push({
+      id: 'service-name',
+      scope: TraceqlSearchScope.Resource,
+      tag: 'service.name',
+      value: serviceName,
+      operator: '=',
+      valueType: 'string',
+    });
   }
   if (spanName !== '') {
-    query.spanName = spanName;
+    query.filters.push({
+      id: 'span-name',
+      scope: TraceqlSearchScope.Span,
+      tag: 'name',
+      value: spanName,
+      operator: '=',
+      valueType: 'string',
+    });
```

Comment:
This sets the service/span name in the tags, as opposed to the dedicated `Resource Service Name` or `Span Name` fields. Users cannot select service/span name in the tags but when it's injected, if they change the tag they cannot reselect either. Also, what happens if they also select service/span name in the dedicated fields.. would the tags override those values.

Reply:
So it's more of an edge case but is possible. I select app in the service graph and then also set the service name myself which results in `resource.service.name` being in the query twice. You can't add `resource.service.name` in the query twice from the editor. We can merge as is this is not a big issue.

![Screenshot 2023-10-09 at 13 34 02](https://github.com/grafana/grafana/assets/90795735/873d4b71-c1a4-40eb-92dd-137623f8ced0)